### DISCUSSION TITLE
cape-file has problems determining file boundaries

### DISCUSSION THREAD

**Discussion by Doodler8888** [minad/corfu#502] [Q&A] [answered]
Suppose i have a file path:

![image](https://github.com/user-attachments/assets/5dc87c0e-0d07-4d81-a1cb-fede3bc45edc)

I want to edit the path by changing the name of some directory inside of it:

![image](https://github.com/user-attachments/assets/7914ec1c-d682-4410-8219-78b2a10581ae)

Then i activate 'completion-at-point' to find a new directory and my cursor ends up at the very start of the path:

![image](https://github.com/user-attachments/assets/4734b70f-cc2f-4b98-a1e5-870ccc12a799)

It noticed that only in org-mode, so maybe it's related solely to it (as usual). But i don't know where to try to report first.

My current config:

```
;; Corfu setup
(use-package corfu
  :ensure t
  :init
  (global-corfu-mode)
  ;; :custom
  ;; (corfu-auto nil)
  ;; (corfu-min-length 2)
  :config
  (advice-add 'pcomplete-completions-at-point :around #'cape-wrap-silent)
  (advice-add 'pcomplete-completions-at-point :around #'cape-wrap-purify)
  (corfu-echo-mode)
  (corfu-history-mode)
  (corfu-popupinfo-mode))

(with-eval-after-load 'corfu
  (define-key corfu-map (kbd "RET") nil))

;; Cape setup
(use-package cape
  :ensure t
  :after corfu
  :init
  (setq completion-at-point-functions
        (list #'cape-file
              #'cape-dabbrev
              #'cape-elisp-block)))
```              

GNU Emacs 30.0.91 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.24.43, cairo version 1.18.2) of 2024-10-06

Org mode version 9.7.11 (release_9.7.11)

**Comment by minad** [comment]
I also see this in other modes. I suspect that the problem lies in `cape-file`, when the boundary of the file name is determined via `bounds-of-thing-at-point`:

https://github.com/minad/cape/blob/63ab1490a14f5a6cbfec29eb5af5ffd12ee49180/cape.el#L429

The `thingatpt` library has a hard time to find proper boundaries when editing paths which are invalid at the time you trigger `completion-at-point`.

EDIT: After looking at this more closely, I am confused why the issue happens. Help investigating the problem would be appreciated.

**Reply by minad** [reply, marked as answer]
@Doodler8888 The problem was that the completion table reported wrong completion boundaries. A potential fix has been added in https://github.com/minad/cape/commit/485d8d4ad5aeb17a0f5d3337e600070276448c54.
